packages/components/src/tabs/index.tsx

```diff
@@ -154,6 +164,22 @@ function Tabs( {
 		setSelectedId,
 	] );
 
+	// In controlled mode, make sure browser focus follows the selected tab if
+	// the selection is changed while a tab is already being focused.
+	useLayoutEffect( () => {
+		if ( ! isControlled ) {
+			return;
+		}
+
+		if (
+			tabsHasFocus &&
+			tabsRef.current?.ownerDocument.activeElement !== null &&
+			selectedId !== tabsRef.current?.ownerDocument.activeElement.id
```

Comment:
It looks like the `tabHasFocus` variable already includes a check about the `activeElement` not being falsy (ie. also not `null`) — is there a reason why the check is repeated here?

If not, we could rewrite this check as

```suggestion
			selectedId !== tabsRef.current?.ownerDocument.activeElement?.id
```